### PR TITLE
QR Codes issue282

### PR DESCRIPTION
**Pull request by easoftwareGit**
### Type of Change

- Adding a new entry
- Editing an existing entry (fixing a typo, bug, issues, etc)
- Updating the documentation

### Checklist

- [X] All writings are my own.
- [X] My entry follows the Codecademy Docs style guide.
- [X] My changes generate no new warnings.
- [X] I have performed a self-review of my own writing and code.
- [X] I have checked my entry and corrected any misspellings.
- [X] I have made corresponding changes to the documentation if needed.
- [X] I have confirmed my changes are not being pushed from my forked `main` branch.
- [X] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [X] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/general/concepts/qr-code/qr-code.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+---
```
Comment (SSwiniarski) [inline content/general/concepts/qr-code/qr-code.md:1]: @easoftwareGit The metadata needs to look like this:
```
---
Title: 'The Title'
Description: 'A brief description'
Subjects:
  - 'A subject name'
  - 'A second subject name'
  - 'An nth subject name'
Tags:
  - 'A tag'
  - 'A second tag'
  - 'An nth tag'
CatalogContent:
  - 'learn-example-course'
  - 'path/example-path'
---
```

Subjects come from [this list](https://github.com/Codecademy/docs/blob/main/documentation/subjects).
Tags come from [this list](https://github.com/Codecademy/docs/blob/main/documentation/tags.md).
For catalog content use [this document](https://github.com/Codecademy/docs/blob/main/documentation/catalog-content.md) as a guide.

File: content/general/concepts/qr-code/qr-code.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+CatalogContent:
+  - 'General'
+  - 'QR Codes'
+  - 'Subjects'
+  - 'Tags'
+---
+
+## What is a QR Code?
+
+A **QR Code** (quick response code) is code that can be read by a computer, just like the standard bar codes that appear on products at retail stores. But the QR Code It is different than the standard bar code. The standard bar code is one dimensional and reads horizontally. A QR Code is a square, with the data enclosed within the square. The three squares in the corners of the QR code tell the QR Code scanning software how the QR Code is orientated, so the scanning software can read the QR Code’s data correctly. [Here](https://www.qr-code-generator.com/qr-code-marketing/qr-codes-basics/) is a good web page detailing how the data is structured in the QR Code.
```
Comment (SSwiniarski) [inline content/general/concepts/qr-code/qr-code.md:13]:
```suggestion
A **QR Code** (quick response code) is code that can be read by a computer, just like the standard bar codes that appear on products at retail stores. But the QR Code is different than the standard bar code. The standard bar code is one dimensional and reads horizontally. A QR Code is a square, with the data enclosed within the square. The three squares in the corners of the QR code tell the QR Code scanning software how the QR Code is orientated, so the scanning software can read the QR Code’s data correctly. [Here](https://www.qr-code-generator.com/qr-code-marketing/qr-codes-basics/) is a good web page detailing how the data is structured in the QR Code.  
```

File: content/general/concepts/qr-code/qr-code.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+## Reading a QR Code
+
+To read a QR Code one does not need a special laser bar code reader. Most camera apps in mobile devices contain the QR Code scanning software. Just open the camera app, focus on the QR Code, and a clickable link appears in the camera app.
+
+## Creating a QR Code
+
+Software is used to create the QR Code. The user enters data to be encoded and how the QR Code will be used. When the user is done entering the data, the software will create the QR Code. There are several QR Code generator websites. Some will generate basic QR Codes for free.
+
+Some QR Codes are static, and some are dynamic. A static QR Code has all the data encoded, and the data will not change. If a data entry error is made during QR Code creation, that error is permanent. A dynamic QR Code fixes the problem. A dynamic QR	Code will retrieve data from the internet, allowing the QR Code owner to edit the data as needed.
```
Comment (SSwiniarski) [inline content/general/concepts/qr-code/qr-code.md:27]:
```suggestion
Some QR Codes are static, and some are dynamic. A static QR Code has all the data encoded, and the data will not change. If a data entry error is made during QR Code creation, that error is permanent. A dynamic QR Code fixes the problem. A dynamic QR Code will retrieve data from the internet, allowing the QR Code owner to edit the data as needed.
```

File: content/general/concepts/qr-code/subjects.md [hunk 4]
```diff
@@ -0,0 +1,4 @@
+General
```
Comment (SSwiniarski) [inline content/general/concepts/qr-code/subjects.md:1]: This is not how subjects are added, you need to remove this file from the PR.

File: content/general/concepts/qr-code/tags.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+Bar Code
```
Comment (SSwiniarski) [inline content/general/concepts/qr-code/tags.md:1]: This is not how tags are added, you need to remove this file from the PR.

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@easoftwareGit I did another review of your changes and left a few more comments. You should really read the [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md#metadata) to understand how the metadata section works.

Also, there was no need for a new PR for these changes, you can edit and commit the changes to the branch you're working on and the changes will appear in the PR. I'm going to close the prior PR, and we can work on this one now.

File: content/general/concepts/qr-code/subjects.md (hunk 4 above)
Comment (SSwiniarski) [inline content/general/concepts/qr-code/subjects.md:1]: You still need to remove this file from the PR.

File: content/general/concepts/qr-code/qr-code.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+CatalogContent:
+  - 'General'
+  - 'QR Codes'
+  - 'Subjects'
+  - 'Tags'
+---
+
+## What is a QR Code?
+
+A **QR Code** (quick response code) is code that can be read by a computer, just like the standard bar codes that appear on products at retail stores. But the QR Code It is different than the standard bar code. The standard bar code is one dimensional and reads horizontally. A QR Code is a square, with the data enclosed within the square. The three squares in the corners of the QR code tell the QR Code scanning software how the QR Code is orientated, so the scanning software can read the QR Code’s data correctly. [Here](https://www.qr-code-generator.com/qr-code-marketing/qr-codes-basics/) is a good web page detailing how the data is structured in the QR Code.
```
Comment (SSwiniarski) [inline content/general/concepts/qr-code/qr-code.md:24]:
```suggestion
A **QR Code** (quick response code) is code that can be read by a computer, just like the standard bar codes that appear on products at retail stores. But the QR Code is different than the standard bar code. The standard bar code is one dimensional and reads horizontally. A QR Code is a square, with the data enclosed within the square. The three squares in the corners of the QR code tell the QR Code scanning software how the QR Code is orientated, so the scanning software can read the QR Code’s data correctly. [Here](https://www.qr-code-generator.com/qr-code-marketing/qr-codes-basics/) is a good web page detailing how the data is structured in the QR Code.  
```

File: content/general/concepts/qr-code/qr-code.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
+---
+Title: 'QR Codes'
```
Comment (SSwiniarski) [inline content/general/concepts/qr-code/qr-code.md:2]: @easoftwareGit The metadata needs to look like this:

```
---
Title: 'The Title'
Description: 'A brief description'
Subjects:
  - 'A subject name'
  - 'A second subject name'
  - 'An nth subject name'
Tags:
  - 'A tag'
  - 'A second tag'
  - 'An nth tag'
CatalogContent:
  - 'learn-example-course'
  - 'path/example-path'
---
```

Subjects come from [this list](https://github.com/Codecademy/docs/blob/main/documentation/subjects).
Tags come from [this list](https://github.com/Codecademy/docs/blob/main/documentation/tags.md).
For catalog content use [this document](https://github.com/Codecademy/docs/blob/main/documentation/catalog-content.md) as a guide.

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@easoftwareGit you marked my comments as resolved, but you didn't address them. Make sure you're changing the file in the branch associated with this PR, not the old closed one. Let me know if you're having trouble understanding what is required.

File: content/general/concepts/qr-code/qr-code.md (hunk 6 above)
Comment (easoftwareGit) [reply content/general/concepts/qr-code/qr-code.md:24]: I do not see what changes you are proposing here. the "QR Code" appears to have the leading and trailing **, and is blue. Is there something else I am missing?

**Comment by easoftwareGit** [comment]
I am using Visual Studio Code and the GitHub plugin.

I did make the changes, and then marked them as resolved. I moved data from
Subjects.md and tags.md into qr-code.md,  and inserted and formatted the
data as you suggested. I then added subjects.md and tags.md to .gitignore.

I saved the file with changes. I then committed the file

I have now deleted subjects.md and tags.md.

File: content/general/concepts/qr-code/qr-code.md (hunk 6 above)
Comment (SSwiniarski) [reply content/general/concepts/qr-code/qr-code.md:24]: @easoftwareGit: "But the QR Code It is different" --> "But the QR Code is different"

**Comment by SSwiniarski** [comment]
@easoftwareGit: Now you need to update the metadata. The items under Subjects should only come from the [subjects.md](https://github.com/Codecademy/docs/blob/main/documentation/subjects.md) file in the documentation folder. Tags should come from the [tags.md](https://github.com/Codecademy/docs/blob/main/documentation/tags.md) file in the documentation folder. Please change the Subjects and Tags in the metadata to reflect this. Also, you should _not_ be editing .gitignore, Please undo those edits.

File: content/general/concepts/qr-code/qr-code.md (hunk 6 above)
Comment (easoftwareGit) [reply content/general/concepts/qr-code/qr-code.md:24]: Got it. Took me several time sto read it, but finally found the extra It.

**Comment by easoftwareGit** [comment]
Done

**Comment by SSwiniarski** [comment]
@easoftwareGit I think you're still not understanding how the metadata works. Let me show you this again:

```
---
Title: 'The Title'
Description: 'A brief description'
Subjects:
  - 'A subject name'
  - 'A second subject name'
  - 'An nth subject name'
Tags:
  - 'A tag'
  - 'A second tag'
  - 'An nth tag'
CatalogContent:
  - 'learn-example-course'
  - 'path/example-path'
---
```
This is how the metadata is supposed to look at the head of your entry. It has a `Subjects:` section and a `Tags:` section. 

There are supposed to be two or three items under each of `Subjects:` and `Tags:`. 

You are supposed to select items that are already listed in the `subjects.md` file, and copy them to your entry under the `Subjects:` section of the metadata. 

The same goes for the `Tags:` section. You look at the items already listed in the `tags.md` file and copy them to your entry under the `Tags:` section of the metadata. 

The `subjects.md` file and the `tags.md` file are there to show you what you can list under the `Subjects:` and `Tags:` section of the Metadata. Most entries do not edit the `subjects.md` file or the `tags.md` file.

While you _can_ add items to the `subjects.md` file, that is extremely rare and what you've added here is not nearly broad enough to count as proper subjects.

You can also add new tags to the `tags.md` file, but you should reduce the redundancy of your added tags, and also list them in the `Tags:` section of the metadata.

Please let me know if this is still unclear to you.

**Comment by easoftwareGit** [comment]
Finally it is clicking in. I will go and make the changes.

**Comment by easoftwareGit** [comment]
Before I submit changes, here is what I have done:
1) updated Subjects and Tags section at the top of qr-code.md
removed my edits from /documentation/subjects.ms and /documentation/tags.md

Thanks again for your help.

**Comment by SSwiniarski** [comment]
@easoftwareGit so far so good. Now you need to edit the metadata so that the list of `Subjects:` are only from items that appear in the `subjects.md` file, and the items in `Tags:` only come from the `tags.md` file, and yo only use `CatalogContent:` as it appears in the `catalog-content.md`.  Those files exist to show you what is acceptable content for those elements of the metadata.

I want to make sure you understand this before you make any further edits. You use the files in `docs/documentation/` to tell you what you can list in the metadata.

**Comment by easoftwareGit** [comment]
OK. Let me say it back to you so we can make sure I am doing it correctly.

For Subjects, Tags and CatalogContent sections in the header of my
qr-code.md file:
I need to scan the matching ____.md file (subjects,md, tags.md or
catalog-content.md), and if there is an item in the ____.md file that I
think should be in the corresponding header section, then I place that word
from the ____.md file into to correct header section in mt qr-codes.md file.

For example: in subjects.md, the first item is "Bash/Shell". That item
should NOT be in my Subjects header section. But the 3rd item "Computer
Science" might be included. "Data Science" is probably NOT included.

So I go through the lists, and pick out the values that I think match my
qr-code.md, and put the matching values in the corresponding sections. If
the item is NOT in the subjects,md, tags.md or catalog-content.md file, it
does NOT go in the corresponding header section.

**Comment by SSwiniarski** [comment]
@easoftwareGit that's right.

**Comment by easoftwareGit** [comment]
I have completed updating the Subjects, Tags and CatalogContent sections in
the header of qr-codes.md. Reading through the contents
of catalog-content.md, I could not find any entries that should be included
in qr-codes.md. Should I delete the CatalogContent section the the header
of qr-codes.md, or should I leave it blank?

**Comment by SSwiniarski** [comment]
@easoftwareGit use the following:

```
CatalogContent:
  - 'paths/code-foundations'
  - 'paths/computer-science'
```

**Comment by SSwiniarski** [comment]
@easoftwareGit It's been two weeks, are you still working on this PR. All we need for the next review is the update to CatalogContent: I mentioned above.

**Comment by easoftwareGit** [comment]
I added the two items to CatalogContent and committed the changes right
after you suggested them.

I was waiting for further instructions

**Comment by easoftwareGit** [comment]
I am out of town for a week. I did add the two suggested items
to CatalogContent and committed the changes.

File: content/general/concepts/qr-code/qr-code.md [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+Description: 'QR Codes and how are they used'
```
Comment (SSwiniarski) [inline content/general/concepts/qr-code/qr-code.md:3]: The discription should be a description of the subject, not of the entry.

**Review by SSwiniarski — APPROVED** [review]
Aside from one comment, should be good for a second review.

**Review by KTom101 — CHANGES_REQUESTED** [review]
@easoftwareGit I added a few changes to this. You will also need to remove the `.gitignore` it shouldn't be touched and there are no changes to the `subjects.md` file. You can also [fix the checks](https://github.com/Codecademy/docs/blob/main/.github/CONTRIBUTING.md#i-submitted-a-pr-but-tests-are-failing-how-do-i-fix-them) in the process. If you have trouble with the checks I can do them on my end just let me know. Please tag me when you're done.

**Comment by easoftwareGit** [comment]
Resolved all the suggestions. I still have two questions

1) How do I remove the .gitignore file correctly? Do I just delete it, or
is there a way to flag it so it is not included?
2) Should I fix the checks AFTER I remove the .gitignore file?

**Comment by KTom101** [comment]
@easoftwareGit You can remove it from the PR by running 
`git checkout <branch> -- <file-name>` See the info linked [here](https://stackoverflow.com/questions/39459467/remove-a-modified-file-from-pull-request#:~:text=Switch%20to%20the%20branch%20from%20which%20you%20created%20the%20pull%20request%3A) for more details.

**Comment by KTom101** [comment]
@easoftwareGit Can you also make the changes I requested? You just marked them as resolved without adding them.

**Comment by easoftwareGit** [comment]
Made the changes.
Got .gitignore removed
Did a commit.
Tried to create a Pull Request and got error message: GraphQL error: A pull
request already exists for easoftwareGit:easoftwareGit/issue282.

I don't have yarn installed, so I was not able to do the checking of the
format. KTom101 said they could do it on their end. Can you do it please?

**Review by KTom101 — APPROVED** [review]

**Comment by KTom101** [comment]
@easoftwareGit Entry is [live](https://www.codecademy.com/resources/docs/general/qr-code) and you're currently listed as anonymous: 

Screenshot 2023-03-18 at 8 59 33 PM

If you want to be listed, follow the following steps:

1.) Add your GitHub Username to your Codecademy Profile Details, you can get there on your desktop by pressing on your profile picture in the upper right-hand corner and selecting "Profile" in the drop-down box:

YourGitHubName

2.) Then log out and log back into Codecademy using your GitHub login:

ReLogIn

> **Note:**  Your email used for Codecademy will have to be listed in your GitHub profile for the login feature to properly work. If you use the same email for both GitHub and Codecademy already, then you're set.

**Comment by easoftwareGit** [comment]
I followed the steps provided by KTom101, and it worked! Saw the article
and my photo as one of the contributors!

I would like to thank all of you for your help on this article. As stated
before, I am dyslexic. I can structure a report, essay or article, I can
structure a paragraph, but it is hard for me to spell everything correctly,
and to catch mistakes that pass the spell checker but are still mistakes.
It is also hard for me to read and comprehend everything correctly, even
after multiple readings.

Thank you for your patience and guidance.

**Comment by SSwiniarski** [comment]
@easoftwareGit Well thank you for bearing with the process. We got a strong entry out of one of our oldest outstanding issues. You should be proud of it!

**Comment by yangc95** [comment]
@easoftwareGit Can we make an additional PR for this entry to edit the metadata? It should be a quick process as we'll consider this a typo.

On lines 5 & 7, let's go ahead and add in those single quotes. Thanks in advance!

> I can also assist with this change. Let me know how you would like to proceed.

**Pull request merged**